src/java.base/share/classes/java/io/BufferedInputStream.java

```diff
@@ -583,4 +585,31 @@ public void close() throws IOException {
             // Else retry in case a new buf was CASed in fill()
         }
     }
+
+    @Override
+    public long transferTo(OutputStream out) throws IOException {
+        Objects.requireNonNull(out, "out");
+        if (lock != null) {
+            lock.lock();
+            try {
+                return implTransferTo(out);
+            } finally {
+                lock.unlock();
+            }
+        } else {
+            synchronized (this) {
+                return implTransferTo(out);
+            }
+        }
+    }
+
+    private long implTransferTo(OutputStream out) throws IOException {
+        if (getClass() == BufferedInputStream.class
+                && ((count - pos) <= 0) && (markpos == -1)) {
```

Comment:
Tiny thing: I think with 3 `&&`s we don't need parenthesis around `((count - pos) <= 0) && (markpos == -1))`. How about to leave leave one around `count - pos`?

Reply:
I was using the exact code proposed by @AlanBateman and I don't dare to touch it. ;-)

Reply:
@AlanBateman Please decide. :-)